### PR TITLE
settings redesign: auth passdb/userdb blocks

### PR DESCRIPTION
**Pull request by sirainen**
DOV-6306 & DOV-6424

On top of https://github.com/dovecot/documentation/pull/826

### CODE REVIEW

File: source/configuration_manual/authentication/user_databases_userdb.rst [hunk 1]
```diff
@@ -97,22 +97,23 @@ Userdb settings
...
+   :values: @strlist
    :seealso: @userdb_override_fields;dovecot_core
 
    Userdb fields (and :ref:`authentication-user_database_extra_fields`)
-   that are used, unless overwritten by the userdb driver. They are in format
-   ``key=value key2=value2 ....`` The values can contain :ref:`%variables
-   <config_variables>`. All %variables used here reflect the state BEFORE the
-   userdb lookup.
+   that are used, unless overwritten by the userdb driver. The values can
+   contain :ref:`%variables <config_variables>`. All %variables used here
```
Comment (cmouse) [inline source/configuration_manual/authentication/user_databases_userdb.rst:105]: should %variable be quoted?

File: source/configuration_manual/authentication/user_databases_userdb.rst [hunk 2]
```diff
@@ -97,22 +97,23 @@ Userdb settings
...
+   reflect the state BEFORE the current userdb lookup, and can refer to
```
Comment (cmouse) [inline source/configuration_manual/authentication/user_databases_userdb.rst:106]: maybe \*\*before\*\* ?

File: source/configuration_manual/authentication/user_databases_userdb.rst [hunk 3]
```diff
@@ -97,22 +97,23 @@ Userdb settings
...
 .. dovecot_core:setting:: userdb_override_fields
-   :values: @string
+   :values: @strlist
 
-   :dovecot_core:ref:`userdb_default_fields`, but instead of providing the
-   default values, these values override what the userdb backend returned.
-   All %variables used here reflect the state AFTER the userdb lookup.
+   Same as :dovecot_core:ref:`userdb_default_fields`, but instead of providing
+   the default values, these values override what the userdb backend returned.
+   All %variables used here reflect the state AFTER the userdb lookup, and can
```
Comment (cmouse) [inline source/configuration_manual/authentication/user_databases_userdb.rst:115]: quote for %variables?

Comment (cmouse) [reply source/configuration_manual/authentication/user_databases_userdb.rst:115]: and maybe \*\*after\*\*?

**Review by cmouse — CHANGES_REQUESTED** [review]

File: source/configuration_manual/authentication/user_databases_userdb.rst (hunk 1 above)
Comment (sirainen) [reply source/configuration_manual/authentication/user_databases_userdb.rst:105]: quoted how? why?

Comment (cmouse) [reply source/configuration_manual/authentication/user_databases_userdb.rst:105]: like \`\`%variables\`\`

Comment (sirainen) [reply source/configuration_manual/authentication/user_databases_userdb.rst:105]: I don't think that's possible inside links?..

Comment (cmouse) [reply source/configuration_manual/authentication/user_databases_userdb.rst:105]: there seems to be few outside

Comment (sirainen) [reply source/configuration_manual/authentication/user_databases_userdb.rst:105]: I think better make them all links. I'll look into that.

**Review by cmouse — APPROVED** [review]

**Pull request merged**